### ISSUE TITLE
Syntax error when referencing `Expr` objects containing IR-specific content in the REPL

### ISSUE THREAD

**Issue by jrevels** [JuliaLang/julia#27546]
Might be REPL-only, haven't checked yet. EDIT: Yup, seems to be REPL-only.

Found while trying to update Cassette's reflection/IR pass code for the latest master:

```julia
julia> Expr(:call, GlobalRef(Base.Math, :float), Core.SlotNumber(2)) # broken
ERROR: syntax: Slot objects should not occur in an AST

julia> x = Any[Expr(:call, GlobalRef(Base.Math, :float), Core.SlotNumber(2))] # works
1-element Array{Any,1}:
 :((Base.Math.float)(_2))

julia> x[1] # broken
ERROR: syntax: Slot objects should not occur in an AST

julia> begin # works
           x = Expr(:call, GlobalRef(Base.Math, :float), Core.SlotNumber(2))
           y = Any[x]
           z = [y[1]]
       end
1-element Array{Expr,1}:
 :((Base.Math.float)(_2))
```

**Comment by JeffBezanson** [comment]
I think this is due to the REPL setting `ans` by calling `eval` on an expression containing the value. We could either use another mechanism there, or disable these errors inside `QuoteNode`.

**Comment by jrevels** [comment]
Fixed by #27562
